### PR TITLE
[CAT-859] Convert Jobs to V2

### PR DESCRIPTION
**Pull request by nateshim-indico**

### CODE REVIEW

**Comment by nateshim-indico** [comment]
Do we want to keep `JobAwaiter`? I was thinking it would be more intuitive to either remove or include the awaiting logic within the jobs client itself so that users will point to a single class when doing any `Job` related queries.

File: Examples/GetPredictionsWithReview/Program.cs [hunk 1]
```diff
@@ -21,8 +21,8 @@ public static async Task Main()
             int submissionId = 91345;
 
             string jobId = await client.Submissions().GenerateSubmissionResultAsync(submissionId);
-            var jobResult = await client.JobAwaiter().WaitReadyAsync<JToken>(jobId, default, default);
-            string jobResultUrl = jobResult.Value<string>("url");
+            var jobResult = await client.Jobs().GetResultAsync(jobId, default, default);
```
Comment (nateshim-indico) [inline Examples/GetPredictionsWithReview/Program.cs:24]: Opted to remove JobAwaiter and put all awaiting logic inside Jobs module.

File: IndicoV2.Abstractions/Jobs/IJobsClient.cs [hunk 2]
```diff
@@ -22,16 +21,7 @@ public interface IJobsClient
...
-        Task<JToken> GetResultAsync(string jobId, CancellationToken cancellationToken = default);
-
-        /// <summary>
-        /// Gets Job's result
-        /// </summary>
-        /// <param name="jobId">Job's Id</param>
-        /// <param name="cancellationToken"><c><see cref="CancellationToken"/></c> for handling cancellation of asynchronous operations.</param>
-        /// <returns>Job result</returns>
-        Task<TResult> GetResultAsync<TResult>(string jobId, CancellationToken cancellationToken = default);
+        Task<string> GetResultAsync(string jobId, TimeSpan checkInterval, CancellationToken cancellationToken = default);
```
Comment (goatrocks) [inline IndicoV2.Abstractions/Jobs/IJobsClient.cs:24]: We shouldn't change the return type or add new non-optional parameters to a function if we can avoid it.

File: IndicoV2/Jobs/JobsClient.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            }
+
+            if (status != JobStatus.SUCCESS)
+            {
+                var failReason = await GetFailureReasonAsync(jobId);
+                throw new JobNotSuccessfulException(status, failReason);
+            }
+
+            var result = await _strawberryShake.Jobs().GetResultAsync(jobId, cancellationToken);
+            return result.Job.Result;
```
Comment (goatrocks) [inline IndicoV2/Jobs/JobsClient.cs:50]: `var json = JsonConvert.DeserializeObject<TResult>(result);` is what we were doing in the previous iteration that let us keep this as a generic function. We should do something like that again here.

File: Examples/GetPredictionsWithReview/Program.cs (hunk 1 above)
Comment (nateshim-indico) [reply Examples/GetPredictionsWithReview/Program.cs]: reverted and marked WaitReadyAsync in JobAwaiter as Obsolete

File: IndicoV2.Abstractions/Jobs/IJobsClient.cs (hunk 2 above)
Comment (nateshim-indico) [reply IndicoV2.Abstractions/Jobs/IJobsClient.cs]: reverted

File: IndicoV2/Jobs/JobsClient.cs (hunk 3 above)
Comment (nateshim-indico) [reply IndicoV2/Jobs/JobsClient.cs]: cool thanks!

File: IndicoV2.Tests/Extensions/SubmissionResult/SubmissionResultAwaiterTests.cs [hunk 4]
```diff
@@ -20,14 +20,14 @@ namespace IndicoV2.Tests.Extensions.SubmissionResult
     public class SubmissionResultAwaiterTests
     {
         private static readonly SubmissionStatus[] _submissionStatusesExceptProcessingAndFailed =
-            Enum.GetValues(typeof(SubmissionStatus)).Cast<SubmissionStatus>().Where(s => s != SubmissionStatus.PROCESSING && s != SubmissionStatus.FAILED).ToArray();
+            Enum.GetValues(typeof(SubmissionStatus)).Cast<SubmissionStatus>().Where(s => s is not SubmissionStatus.PROCESSING and not SubmissionStatus.FAILED).ToArray();
```
Comment (goatrocks) [inline IndicoV2.Tests/Extensions/SubmissionResult/SubmissionResultAwaiterTests.cs:23]: nice LINQ use

**Review by goatrocks — APPROVED** [review]

**Pull request merged**